app/javascript/controllers/area_served_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+    static targets= ["share", "total", "warning"]
+    connect () {
+        this.calculateClientShareTotal()
+    }
+    calculateClientShareTotal(){
+        let total = 0;
+        let share_targets = this.shareTargets
+        share_targets.forEach( share_target =>{
```

Comment:
The whitespace formatting/indent of this whole file seems a bit off - please switch to 2-spaces and on line-10 here add in a newline I think. [this is not a required change, but if you're in here anyway....]